service/kernel/oneapi/cl_kernels/select_indexed.cl

```diff
@@ -46,86 +46,102 @@ void partition( __global  algorithmFPType *pdist,
+    if(need_init_labels)
+    {
+        for(int i = start + local_id; i < end; i += local_size)
+        {
+            plabels[i] = i;
+        }
+    }
     for(int i = start + local_id; i < end; i += local_size) 
     {
         sub_group_barrier(CLK_GLOBAL_MEM_FENCE);
         algorithmFPType curVal = pdist[i];
-        int curLabel = plabels[i];
-        unsigned char flag = curVal < pivot ? 1 : 0;
-        unsigned char big_num = sub_group_reduce_add(curVal > pivot ? 1 : 0);
-        unsigned char incr = sub_group_reduce_add(flag);
-        int min_ind = sub_group_reduce_min(i);
+        int curLabel           = plabels[i];
+        unsigned char flag     = curVal < pivot ? 1 : 0;
+        unsigned char big_num  = sub_group_reduce_add(curVal > pivot ? 1 : 0);
+        unsigned char incr     = sub_group_reduce_add(flag);
+        int min_ind            = sub_group_reduce_min(i);
         if(incr > 0) 
         {
-            unsigned char shift = sub_group_scan_exclusive_add(flag);
-            unsigned char old_shift= sub_group_scan_exclusive_add(flag > 0 ? 0 : 1);
-            algorithmFPType exVal = flag > 0 ? pdist[start + *split + shift] : -1.0;
-            int exLabel = flag > 0 ? plabels[start + *split + shift] : -1;
-            int cur_size = i > full_group_size - 1 ? last_group_size : local_size;
+            unsigned char shift     = sub_group_scan_exclusive_add(flag);
+            unsigned char old_shift = sub_group_scan_exclusive_add(flag > 0 ? 0 : 1);
+            int cur_size            = i > full_group_size - 1 ? last_group_size : local_size;
             if(flag) 
             {
-                pdist[start + *split + shift] = curVal;
+                algorithmFPType exVal           = pdist[start + *split + shift];
+                int exLabel                     = plabels[start + *split + shift];
+                pdist[start + *split + shift]   = curVal;
                 plabels[start + *split + shift] = curLabel;
-                if(start + *split + shift < min_ind) {
-                    pdist[min_ind + cur_size - 1 - shift] = exVal;
+                if(start + *split + shift < min_ind) 
+                {
+                    pdist[min_ind + cur_size - 1 - shift]   = exVal;
                     plabels[min_ind + cur_size - 1 - shift] = exLabel;
                 }
             } 
             else 
             {
-                pdist[min_ind + cur_size - 1 - old_shift] = curVal;
+                pdist[min_ind + cur_size - 1 - old_shift]   = curVal;
                 plabels[min_ind + cur_size - 1 - old_shift] = curLabel;
             }
-        }
+        } 
         sub_group_barrier(CLK_GLOBAL_MEM_FENCE);
-        *split += incr;
+        *split         += incr;
         *total_big_num += big_num;
     }
-    *split = - sub_group_reduce_min(-(*split));
+    *split         = - sub_group_reduce_min(-(*split));
     *total_big_num = - sub_group_reduce_min(-(*total_big_num));
 }
```

Comment:
I am sorry, but I do not understand what is happening inside this function. Maybe we should better name some of elements:
1. Name of the function. What we are partitioning? And what strategy we use? The name should reflect the purpose of the function.
2. `start`, `end` - do not see what the array they are pointed to. It can be detected by reading the code, but it would be better to reflect the `physical` meaning of this pointers in their names
3. For me, it absolutely meaningless names of `pivot`, `split`, `flag`, `bigNum`, `incr` and so on. 
4. Please, do not shorten variable names like `exLabel`.